Nodejs/Product/TestAdapter/TestFrameworks/Jest/jest.js

```diff
@@ -51,14 +53,15 @@ const run_tests = function (context) {
             });
         }
 
-        const config = {
+        let projectFolderJsonReady = projectFolder.replaceAll('\\', '\\\\');
+        const argv = {
             json: true,
             reporters: [[__dirname + '/jestReporter.js', { context }]],
-            testMatch: [context.testCases[0].testFile]
-        };
+            config : `{"roots":["<rootDir>/src"],"collectCoverageFrom":["src/**/*.{js,jsx,ts,tsx}","!src/**/*.d.ts"],"setupFiles":["${projectFolderJsonReady}\\\\node_modules\\\\react-app-polyfill\\\\jsdom.js"],"setupFilesAfterEnv":["<rootDir>/src/setupTests.js"],"testMatch":["<rootDir>/src/**/__tests__/**/*.{js,jsx,ts,tsx}","<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}"],"testEnvironment":"jsdom","testRunner":"${projectFolderJsonReady}\\\\node_modules\\\\jest-circus\\\\runner.js","transform":{"^.+\\\\.(js|jsx|mjs|cjs|ts|tsx)$":"${projectFolderJsonReady}\\\\node_modules\\\\react-scripts\\\\config\\\\jest\\\\babelTransform.js","^.+\\\\.css$":"${projectFolderJsonReady}\\\\node_modules\\\\react-scripts\\\\config\\\\jest\\\\cssTransform.js","^(?!.*\\\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)":"${projectFolderJsonReady}\\\\node_modules\\\\react-scripts\\\\config\\\\jest\\\\fileTransform.js"},"transformIgnorePatterns":["[/\\\\\\\\]node_modules[/\\\\\\\\].+\\\\.(js|jsx|mjs|cjs|ts|tsx)$","^.+\\\\.module\\\\.(css|sass|scss)$"],"modulePaths":[],"moduleNameMapper":{"^react-native$":"react-native-web","^.+\\\\.module\\\\.(css|sass|scss)$":"identity-obj-proxy"},"moduleFileExtensions":["web.js","js","web.ts","ts","web.tsx","tsx","json","web.jsx","jsx","node"],"watchPlugins":["jest-watch-typeahead/filename","jest-watch-typeahead/testname"],"resetMocks":true,"rootDir":"${projectFolderJsonReady}"}`,
```

Comment:
This is a little hard to read. The usual approach to this is to first create the json object then stringify it:

```
const config = { ... };
const argv = {
  ...
  config: JSON.stringify(config)
};
```

As a bonus this lets you get rid of the `replaceAll`.

Reply:
We might have to find a minimal set of options. It doesn't look like some of these should have to be here. Also looking at the paths, I see a lot of "/src/" folders - while this might be a convention, I don't think this is a required project structure for react tests to work (what if I structured my code to be a "source/" directory instead for example).

Reply:
I'm a little unsure what to do with the "/src", I agree with @armanio123 that we need to suit this accordingly if there's already a config file on the project's folder, but how can we handle the fact that the user can change the name of the folder and maybe not have a config file? (Or am I understanding it wrong? That can also be the case :D ) Because from my tests creating a React app, it does create the folder "/src".

Reply:
After new changes, we're not configuring this by hand anymore, we're letting `react-scripts` give us the config and, if not, adding the one from the user or passing just the `testMatch` config that we were doing previously.